packet/commands.py

```diff
@@ -141,30 +156,64 @@ def ldap_sync():
+    webmaster = ldap_get_webmaster()
+    c_m = ldap_get_constitutional_maintainer()
+    drink = ldap_get_drink_admin()
 
     print("Applying updates to the DB...")
     for packet in Packet.query.filter(Packet.end > datetime.now()).all():
-        # Update the eboard state of all UpperSignatures
+        # Update the role state of all UpperSignatures
         for sig in filter(lambda sig: sig.member in all_upper, packet.upper_signatures):
-            sig.eboard = ldap_is_eboard(all_upper[sig.member])
+            sig.eboard = ldap_get_eboard_role(all_upper[sig.member])
+            sig.active_rtp = sig.member in rtp
+            sig.three_da = sig.member in three_da
+            sig.webmaster = sig.member in webmaster
+            sig.c_m = sig.member in c_m
+            sig.drink_admin = sig.member in drink
 
         # Migrate UpperSignatures that are from accounts that are not active anymore
         for sig in filter(lambda sig: sig.member not in all_upper, packet.upper_signatures):
             UpperSignature.query.filter_by(packet_id=packet.id, member=sig.member).delete()
             if sig.signed:
-                db.session.add(MiscSignature(packet=packet, member=sig.member))
+                sig = MiscSignature(packet=packet, member=sig.member)
+                db.session.add(sig)
+
+        # Migrate UpperSignatures that are from accounts that are on co-op currently
+        for sig in filter(lambda sig: sig.member in on_coop, packet.upper_signatures):
+            UpperSignature.query.filter_by(packet_id=packet.id, member=sig.member).delete()
+            if sig.signed:
+                sig = MiscSignature(packet=packet, member=sig.member)
+                db.session.add(sig)
 
         # Migrate MiscSignatures that are from accounts that are now active members
-        for sig in filter(lambda sig: sig.member in all_upper, packet.misc_signatures):
+        for sig in filter(lambda sig: sig.member in all_upper and not on_coop, packet.misc_signatures):
             MiscSignature.query.filter_by(packet_id=packet.id, member=sig.member).delete()
-            db.session.add(UpperSignature(packet=packet, member=sig.member,
-                                          eboard=ldap_is_eboard(all_upper[sig.member]), signed=True))
+            sig = UpperSignature(packet=packet, member=sig.member, signed=True)
+            sig.eboard = ldap_get_eboard_role(all_upper[sig.member])
+            sig.active_rtp = sig.member in rtp
+            sig.three_da = sig.member in three_da
+            sig.webmaster = sig.member in webmaster
+            sig.c_m = sig.member in c_m
+            sig.drink_admin = sig.member in drink
+            db.session.add(sig)
 
         # Create UpperSignatures for any new active members
         # pylint: disable=cell-var-from-loop
         upper_sigs = set(map(lambda sig: sig.member, packet.upper_signatures))
         for member in filter(lambda member: member not in upper_sigs, all_upper):
```

Comment:
Might want to check against the co_op list here

Reply:
Oh wait I think I tried that, I can't remember why I removed that check

Reply:
I think the edge case is here. If a new active member is marked as on coop, this loop won't handle that.